src/app/data-model-interface/DataModel.h

```diff
@@ -55,22 +55,24 @@ class DataModel : public DataModelMetadataTree
     // event emitting, path marking and other operations
     virtual InteractionModelContext CurrentContext() const { return mContext; }
 
-    /// List reading has specific handling logic:
-    ///   `state` contains in/out data about the current list reading. MUST start with kInvalidListIndex on first call
-    ///
     /// Return codes:
-    ///   CHIP_ERROR_MORE_LIST_DATA_AVAILABLE (NOTE: new error defined for this purpose)
-    ///      - partial data written to the destination
-    ///      - destination will contain AT LEAST one valid list entry fully serialized
-    ///      - destination will be fully valid (it will be rolled back on partial list writes)
+    ///   CHIP_ERROR_ACCESS_DENIED:
+    ///      - May be ignored if reads use path expansion (e.g. to skip inaccessible attributes
+    ///        during subscription requests). This code MUST be used for ACL failures and only for ACL failures.
+    ///   CHIP_IM_GLOBAL_STATUS(UnsupportedRead):
+    ///      - May be ignored if reads use path expansion (e.g. to skip reading write-only attributes
+    ///        during subscription requests).
```

Comment:
This one is a bit better, perhaps, since it's sort of existing API surface for AttributeAccessInterface, but still not very reliable.....